pkg/apis/hypercloud/v1alpha1/virtualmachinevolumeexport_types.go

```diff
@@ -0,0 +1,88 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// VirtualMachineVolumeExportSpec defines the desired state of VirtualMachineVolumeExport
+type VirtualMachineVolumeExportSpec struct {
+	VirtualMachineVolume VirtualMachineVolumeSource            `json:"virtualMachineVolume"`
+	Destination          VirtualMachineVolumeExportDestination `json:"destination"`
+}
+
+// VirtualMachineVolumeExportDestination defines the destination to export volume
+type VirtualMachineVolumeExportDestination struct {
+	Local *VirtualMachineVolumeExportDestinationLocal `json:"local,omitempty"`
+	S3    *VirtualMachineVolumeExportDestinationS3    `json:"s3,omitempty"`
+}
+
+// VirtualMachineVolumeSource indicates the VirtualMachineVolume to be exported
+type VirtualMachineVolumeSource struct {
+	Name string `json:"name"`
+}
+
+// VirtualMachineVolumeExportDestinationLocal defines the Local destination to export volume
+type VirtualMachineVolumeExportDestinationLocal struct{}
+
+// VirtualMachineVolumeExportDestinationS3 defines the S3 destination to export volume
+type VirtualMachineVolumeExportDestinationS3 struct {
+	URL    string `json:"url,omitempty"`
+	Secret string `json:"secretRef,omitempty"`
+}
+
+// VirtualMachineVolumeExportStatus defines the observed state of VirtualMachineVolumeExport
+type VirtualMachineVolumeExportStatus struct {
+	// State is the current state of VirtualMachineVolumeExport
+	State VirtualMachineVolumeExportState `json:"state"`
```

Comment:
We've decided to use the condition array. Please open the issue to resolve it in another commit

Reply:
Done.